utils/generate-api.py

```diff
@@ -637,6 +642,11 @@ def read_modules():
                 paths.append({"path": key2, "methods": methods})
 
         api.update({"url": {"paths": paths}})
+        if all_paths_have_deprecation and x_deprecation_message is not None:
+            api.update({"deprecation_message": x_deprecation_message})
+
+        if namespace == "indices" and name == "put_mapping":
```

Comment:
Do you want to try to extract the JSON patch logic from my other PR after this, and we can discuss whether it's a good idea, or do you want me to do it?

Reply:
https://github.com/opensearch-project/opensearch-py/pull/553

Reply:
Consolidating all patches in one place within overrides is efficient. However, creating a completely new spec for those endpoints might risk becoming outdated and misaligned with the main spec.